library/std/src/path.rs

```diff
@@ -2463,7 +2498,16 @@ impl Path {
     /// This is a convenience function that coerces errors to false. If you want to
     /// check errors, call [`fs::metadata`] and handle its [`Result`]. Then call
     /// [`fs::Metadata::is_dir`] if it was [`Ok`].
+    ///
+    /// When the goal is simply to read from the source, the most reliable way to
+    /// test the source can be read is to open it. See [`fs::read_dir`] for more
+    /// information.
```

Comment:
I don't think we need to update the deprecated docs.

Reply:
I added this because `is_file` already has a similar comment, and these comments relate to one of the reasons for deprecating these functions, so it seems like useful context.